pkg/scheduler/objects/queue.go

```diff
@@ -624,12 +624,19 @@ func (sq *Queue) CheckAdminAccess(user security.UserGroup) bool {
 }
 
 // GetPartitionQueueDAOInfo returns the queue hierarchy as an object for a REST call.
-func (sq *Queue) GetPartitionQueueDAOInfo() dao.PartitionQueueDAOInfo {
+// Exclude is true, which means that returns the specified queue object, but does not return the children of the specified queue.
+func (sq *Queue) GetPartitionQueueDAOInfo(exclude bool) dao.PartitionQueueDAOInfo {
 	queueInfo := dao.PartitionQueueDAOInfo{}
-	childes := sq.GetCopyOfChildren()
-	queueInfo.Children = make([]dao.PartitionQueueDAOInfo, 0, len(childes))
-	for _, child := range childes {
-		queueInfo.Children = append(queueInfo.Children, child.GetPartitionQueueDAOInfo())
+	children := sq.GetCopyOfChildren()
+	queueInfo.Children = make([]dao.PartitionQueueDAOInfo, 0, len(children))
+	if exclude {
+		for _, child := range children {
+			queueInfo.ChildrenNames = append(queueInfo.ChildrenNames, child.QueuePath)
+		}
+	} else {
+		for _, child := range children {
+			queueInfo.Children = append(queueInfo.Children, child.GetPartitionQueueDAOInfo(false))
+		}
```

Comment:
refactor: only need to get a copy of children if we need to recurse into them:
```
if !exclude {
	children := sq.GetCopyOfChildren()
	queueInfo.Children = make([]dao.PartitionQueueDAOInfo, 0, len(children))
	for _, child := range children {
		queueInfo.Children = append(queueInfo.Children, child.GetPartitionQueueDAOInfo(false))
	}
}
```
Leave adding the child path names to the locked code below

Reply:
Add this:
```
for _, child := range children {
	queueInfo.ChildrenNames = append(queueInfo.ChildrenNames, child.QueuePath)
}
```
around properties for loop line 660